src/middleware/transactionBody.js

```diff
@@ -0,0 +1,26 @@
+import mongodb from 'mongodb'
+
+export const retrieveTransactionBody = (db, fileId, callback) => {
+  if (!db) {
+      const err = new Error(`Transaction body retrieval failed. Database handle: ${db} is invalid`)
+      return callback(err, null)
+  }
+
+  if (!fileId) {
+    const err = new Error(`Transaction body retrieval failed: Transaction id: ${fileId}`)
+    return callback(err, null)
+  }
+
+  const bucket = new mongodb.GridFSBucket(db)
+
+  let body = ''
+  bucket.openDownloadStream(fileId)
+    .on('error', err => {
+      const error = new Error(`Transaction body retrieval failed: Error in reading stream: ${err.message}`)
```

Comment:
The description also makes reference to Transaction which it shouldnt be, as this function could be used for any GridFS document retrieval